app/src/main/res/layout/item_search_result.xml

```diff
@@ -48,9 +48,26 @@
             android:ellipsize="end"
             android:maxLines="1"
             app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/delete"
```

Comment:
To keep the same space between the end of the text and start of the icon if you use the insets, you'll need to modify the horizontal margin so it keeps 16dp at the start but has only 8dp at end because of the insets that already provide 8dp by themselves